### PR TITLE
Rocketchat plugin

### PR DESCRIPTION
**Pull request by dhollinger**

### CODE REVIEW

File: lib/plugins/chatops.rb [hunk 1]
```diff
@@ -21,6 +21,14 @@ def notify(message)
                            message,
                            http_options: @args[:http_options] || {},
                            icon_emoji: @args[:icon_emoji]).notify
+      when 'rocketchat'
+        LOGGER.info("Sending Rocket.Chat webhook message to #{@url}")
+        Chatops::Rocketchat.new(@channel,
```
Comment (bastelfreak) [inline lib/plugins/chatops.rb]: Personally I prefer it when all params in new() are in the next line, limits the indentation to one level

Comment (dhollinger) [reply lib/plugins/chatops.rb]: I believe that is against the Ruby Style Guide and I don't think we turned off those cops in Rubocop. I saw failures related to trying to put parameters in-between open/closing parentheses on their own lines. It wants the first and last parameters on the same line as the open/close.

Same goes when calling a class method after the `.new()`, it wants the final parameter, closing parenthesis, and method call all on the same line.

Comment (dhollinger) [reply lib/plugins/chatops.rb]: I found a solution that should satisfy you, @bastelfreak. I'm pretty sure it's close to what you are asking.

File: README.md [hunk 2]
```diff
@@ -104,7 +104,8 @@ Then configure the webhook to add your Slack Webhook URL:
 ``` yaml
 chatops: true
 chatops_service: 'slack' # Required so the app knows that you're sending to Slack.
-chatops_channel: '#channel' # deftaults to #general
+chatops_url: 'http://slack.webhook/webhook' # mandatory for usage
```
[the remark this answers is not in the document]
Comment (dhollinger) [reply README.md]: True, but slack is always `https://hooks.slack.com/<hash>/<hash>/<hash>`

File: README.md [hunk 3]
```diff
@@ -116,6 +117,35 @@ chatops_options:
 
 **NOTE: The legacy `slack_webhook`, `slack_user`, `slack_channel`, `slack_emoji`, and `slack_proxy_url` still work, but will be removed in 3.0.0**
 
+##### Rocket.Chat Configuration
+
+You can enable Rocket.Chat notifications for the webhook. You will need a
+Rocket.Chat incoming webhook URL and the `rocket-chat-notifier` gem installed.
```
[the remark this answers is not in the document]
Comment (dhollinger) [reply README.md]: As a runtime dependency, the `rocket-chat-notifier` gem will be installed as a dependency. We could make the gem "optional", but the more you do that, the more work you put on your users to install those "optional" dependencies themselves. 

Think of `puppetlabs-beaker` dependencies, there are so many dependencies for each supported hypervisor that you either install a ton of deps during gem install or put the entire onus on the user to install what he needs (though he may not know what he needs). 

I think at this point, we should leave it as a runtime dependency and instead make it optional in the future by providing the ChatOps plugin as a separate gem for puppet_webhook unto itself.

File: lib/plugins/chatops.rb [hunk 4]
```diff
@@ -15,12 +15,14 @@ def notify(message)
...
       when 'slack'
         LOGGER.info("Sending Slack webhook message to #{@url}")
-        Chatops::Slack.new(@channel,
-                           @url,
-                           @user,
-                           message,
-                           http_options: @args[:http_options] || {},
-                           icon_emoji: @args[:icon_emoji]).notify
+        Chatops::Slack.new(
+          @channel, @url, @user, message, http_options: @args[:http_options] || {}, icon_emoji: @args[:icon_emoji]
```
[the remark this answers is not in the document]
Comment (dhollinger) [reply lib/plugins/chatops.rb]: I'll let you and @bastelfreak fight over this one 😃

File: lib/plugins/chatops.rb [hunk 5]
```diff
@@ -15,12 +15,14 @@ def notify(message)
       case @service
       when 'slack'
         LOGGER.info("Sending Slack webhook message to #{@url}")
-        Chatops::Slack.new(@channel,
-                           @url,
-                           @user,
-                           message,
-                           http_options: @args[:http_options] || {},
-                           icon_emoji: @args[:icon_emoji]).notify
+        Chatops::Slack.new(
```
Comment (binford2k) [inline lib/plugins/chatops.rb]: You might consider moving the `require 'plugins/chatops/slack'` in here. That will only load the integration you want to use. Not a big deal now, but as you start adding more integrations, you might not want them all loaded all the time.

File: lib/plugins/chatops.rb [hunk 6]
```diff
@@ -15,12 +15,14 @@ def notify(message)
...
+          @channel, @url, @user, message, http_options: @args[:http_options] || {}, icon_emoji: @args[:icon_emoji]
+        ).notify
+      when 'rocketchat'
```
Comment (binford2k) [inline lib/plugins/chatops.rb]: Don't forget to `require 'plugins/chatops/rocketchat'`

Comment (dhollinger) [reply lib/plugins/chatops.rb]: I knew I forgot something

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhollinger) [reply README.md]: @rnelson0 I should probably add that, right now, the Chatops class doesn't test to see if a dependency exists before loading it, so it expects all dependencies to be there.

File: lib/plugins/chatops.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (bastelfreak) [reply lib/plugins/chatops.rb]: oh sorry I wasn't clear. I meant something like this:

```ruby
Chatops::Slack.new(
  bla,
  bla2,
)
```

that's also enforced via rubocop in our modules, but depending on the current code it doesn't detect it / accepts different styles as well.

[the remark this answers is not in the document]
Comment (dhollinger) [reply lib/plugins/chatops.rb]: ahh, ok. I fixed that

**Pull request merged**